src/atomate2/abinit/jobs/base.py

```diff
@@ -0,0 +1,815 @@
+            a dict with keys being a type of previous output and values being one of the previous (absolute
+            path or list of absolute paths, uuid or list of uuids, ...), ...
+        restart_from : TBD
+            An absolute path as a string (previous directory), a string (previous job uuid), an OutputReference,
+            TODO: what else ? is an AbinitInputSet ok ?
+            TODO: I would say no, because we not only need the input but also the location where it previously ran.
+        """
+        if structure is None and prev_outputs is None and restart_from is None:
+            raise RuntimeError(
+                "At least one of structure, prev_outputs or restart_from should be defined."
+            )
+
+        self.setup_job(
+            structure=structure,
+            prev_outputs=prev_outputs,
+            restart_from=restart_from,
+            # )
+            # abinit_input_set=abinit_input_set,
+            # previous_abinit_input_set=previous_abinit_input_set,
+            # restart_info=restart_info,
+            history=history,
+        )
+        self.run_abinit()
+        response = self.job_analysis()
+
+        return response
+
+    def setup_job(
+        self,
+        structure: Optional[Structure] = None,
+        prev_outputs: Optional[Any] = None,
+        restart_from=None,
+        # restart_info: Optional[RestartInfo] = None,
+        # ):
+        # previous_abinit_input_set: Optional[AbinitInputSet] = None,
+        # restart_info: Optional[RestartInfo] = None,
+        history: Optional[JobHistory] = None,
+    ):
+        """Set up abinit job.
+
+        Parameters
+        ----------
+        structure
+        prev_outputs
+        restart_from
+        """
+        self.start_time = time.time()
+
+        self.structure = structure
+        # TODO: transform restart_from and prev_outputs here so that
+        #  they are in the format expected by the AbinitInputSet
+        self.restart_from = restart_from
+        # The previous outputs can be a single previous output, a list of previous outputs,
+        # a dict with values being previous outputs.
+        self.prev_outputs = prev_outputs
+        if isinstance(prev_outputs, AbinitJobSummary):
+            self.prev_outputs = [prev_outputs]
+        elif isinstance(prev_outputs, list):
+            self.prev_outputs = prev_outputs
+        elif prev_outputs is None:
+            self.prev_outputs = prev_outputs
+        else:
+            raise NotImplementedError()
+        # TODO: see if we keep this restart_info or if we just use the history
+        self.history = history or JobHistory()
+
+        # Set up logging
+        self.set_logger()
+
+        # Load the atomate settings for abinit to get configuration parameters
+        # TODO: how to allow for tuned parameters on a per-job basis ? (similar to fw_spec-passed settings)
+        self.settings = AbinitAtomateSettings()
```

Comment:
You should put all abinit settings in the normal Atomate2Settings object. Then you can just import `from atomate2 import SETTINGS` to have access to the currently loaded settings.

For fine tuning parameters, these should be controlled through settings on the maker. E.g., see the job makers in the VASP module.

Reply:
done